mutiny-core/src/federation.rs

```diff
@@ -238,15 +263,44 @@ impl FederationClient {
         Ok(self.fedimint_client.get_balance().await.msats / 1_000)
     }
 
-    pub async fn get_activity(&self) -> Result<Vec<ActivityItem>, MutinyError> {
+    pub async fn check_activity(&self) -> Result<(), MutinyError> {
         log_trace!(self.logger, "Getting activity");
-        let payments = self.g.list_payments().await?;
 
-        let mut payments_map: HashMap<sha256::Hash, MutinyInvoice> = HashMap::new();
+        let now = utils::now();
+        let mut payments = Vec::new();
+        // inbound
+        payments.extend(
+            list_payment_info(self.storage.clone(), true)?
+                .into_iter()
+                .filter_map(|(h, i)| {
+                    let mutiny_invoice = MutinyInvoice::from(i.clone(), h, true, vec![]).ok();
+
+                    // filter out expired invoices
+                    mutiny_invoice.filter(|invoice| {
+                        !invoice.bolt11.as_ref().is_some_and(|b| b.would_expire(now))
+                            || matches!(i.status, HTLCStatus::Succeeded | HTLCStatus::InFlight)
+                    })
+                }),
+        );
+        // outbound
+        payments.extend(
+            list_payment_info(self.storage.clone(), false)?
+                .into_iter()
+                .filter_map(|(h, i)| {
+                    // TODO labels
+                    let mutiny_invoice = MutinyInvoice::from(i.clone(), h, false, vec![]).ok();
+
+                    // filter out expired invoices
+                    mutiny_invoice.filter(|invoice| {
+                        !invoice.bolt11.as_ref().is_some_and(|b| b.would_expire(now))
+                            || matches!(i.status, HTLCStatus::Succeeded | HTLCStatus::InFlight)
+                    })
+                }),
+        );
```

Comment:
this seems suspect to me

The `.filter` seems wrong to me, we don't care about successful payments because those are already completed. I think you'd want to get `InFlight` and `Pending` here, also don't want to filter out expired invoices because we should check if they've been paid in the mean time, if not then we should set to failed.

I assume this will effect the logic below

Reply:
Good catch, I now just filter out `Succeeded` and `InFlight`. Testing it now but all the other code changes have been handled.